ocaml/xapi/xapi_vm.ml

```diff
@@ -261,9 +261,19 @@ let start ~__context ~vm ~start_paused ~force =
     end else vmr
   in
 
+  (* Check to see if secure boot should be parsed *)
+  let pool = Helpers.get_pool ~__context in
+  if List.assoc "secureboot" platform = "auto" then
+    Db.VM.set_platform ~__context ~self:vm
```

Comment:
I'm not sure but maybe "add_to_platform" could be also used - I don't know how it handles existing entries.

Reply:
The `platform` field is set just above here as well. It would also be better to avoid doing it twice for the same field, since we are potentially running on a slave.

Reply:
I think
`if List.mem_assoc key t && (not idempotent || List.assoc key t <> value) then raise Duplicate;`
If the key already exists, and either idempotent is true, or the current value is different to the new value - then raise duplicate. So it reads to me that add_to_platform wouldn't work well for existing keys?

Reply:
I agree. Keep the current code.